### PR TITLE
Fixes "VSTS 524616 - can't refresh the website successfully after del…

### PR DESCRIPTION
**Pull request by mkrueger**
…ete the content in '.cshtml' file"

https://devdiv.visualstudio.com/DevDiv/_workitems?id=524616
Was caused by an incompatibility of the vs text buffer with our text buffer due to the changed event behavior.

### CODE REVIEW

File: main/src/core/MonoDevelop.TextEditor.Tests/Mono.TextEditor.Tests/DocumentTests.cs [hunk 1]
```diff
@@ -195,5 +195,21 @@ public void TestBug53380 ()
...
+		/// VSTS 524616 can't refresh the website successfully after delete the content in '.cshtml' file
+		/// </summary>
+		[Test]
+		public void TestVSTS524616 ()
+		{
+			var document = new Mono.TextEditor.TextDocument ();
+			document.Text = "test";
+			string txt;
+			document.TextChanging += delegate {
+				txt = document.Text;
```
Comment (Therzok) [inline main/src/core/MonoDevelop.TextEditor.Tests/Mono.TextEditor.Tests/DocumentTests.cs]: txt is not used.

Comment (KirillOsenkov) [reply main/src/core/MonoDevelop.TextEditor.Tests/Mono.TextEditor.Tests/DocumentTests.cs]: In fact the entire delegate is not used.

**Comment by KirillOsenkov** [comment]
Also the failure only used to happen in old Razor documents as far as I can tell - the test doesn't seem to use that?

File: main/src/core/MonoDevelop.TextEditor.Tests/Mono.TextEditor.Tests/DocumentTests.cs (hunk 1 above)
Comment (mkrueger) [reply main/src/core/MonoDevelop.TextEditor.Tests/Mono.TextEditor.Tests/DocumentTests.cs]: not needed to be used it should only be called.

**Comment by mkrueger** [comment]
y the test fails I write tests before the fix ... ty for asking

**Comment by mkrueger** [comment]
The failure is not really connected to razor but the old razor code does what the unit test does. I don't plan to use the razor code in any test if it's not needed. The old razor code should go away IMO.

We're using the VS.NET webtools AFAIK - sooner or later we'll get a better razor implementation.

**Pull request merged**